store/storetest/file_info_store.go

```diff
@@ -256,6 +258,133 @@ func testFileInfoGetForUser(t *testing.T, ss store.Store) {
 	assert.Len(t, userPosts, 1)
 }
 
+func testFileInfoGetWithOptions(t *testing.T, ss store.Store) {
+
+	makePost := func(chId string, user string) model.Post {
+		post := model.Post{}
+		post.ChannelId = chId
+		post.UserId = user
+		_, err := ss.Post().Save(&post)
+		require.Nil(t, err)
+		return post
+	}
+
+	makeFile := func(post model.Post, user string, createAt int64, idPrefix string) model.FileInfo {
+		id := model.NewId()
+		id = idPrefix + id[1:] // hacky way to get sortable Ids to confirm secondary Id sort works
+		fileInfo := model.FileInfo{
+			Id:        id,
+			CreatorId: user,
+			Path:      "file.txt",
+			CreateAt:  createAt,
+		}
+		if post.Id != "" {
+			fileInfo.PostId = post.Id
+		}
+		_, err := ss.FileInfo().Save(&fileInfo)
+		require.Nil(t, err)
+		return fileInfo
+	}
+
+	userId1 := model.NewId()
+	userId2 := model.NewId()
+
+	channelId1 := model.NewId()
+	channelId2 := model.NewId()
+	channelId3 := model.NewId()
+
+	post1_1 := makePost(channelId1, userId1) // post 1 by user 1
+	post1_2 := makePost(channelId3, userId1) // post 2 by user 1
+	post2_1 := makePost(channelId2, userId2)
+	post2_2 := makePost(channelId3, userId2)
+
+	epoch := time.Date(2020, 1, 1, 1, 1, 1, 1, time.UTC)
+	file1_1 := makeFile(post1_1, userId1, epoch.AddDate(0, 0, 1).Unix(), "a")      // file 1 by user 1
+	file1_2 := makeFile(post1_2, userId1, epoch.AddDate(0, 0, 2).Unix(), "b")      // file 2 by user 1
+	file1_3 := makeFile(model.Post{}, userId1, epoch.AddDate(0, 0, 3).Unix(), "c") // file that is not attached to a post
+	file2_1 := makeFile(post2_1, userId2, epoch.AddDate(0, 0, 4).Unix(), "d")      // file 2 by user 1
+	file2_2 := makeFile(post2_2, userId2, epoch.AddDate(0, 0, 5).Unix(), "e")
+
+	// delete a file
+	_, err := ss.FileInfo().DeleteForPost(file2_2.PostId)
+	require.Nil(t, err)
+
+	testCases := []struct {
+		Name            string
+		Page, PerPage   uint
+		Opt             *model.GetFileInfosOptions
+		ExpectedFileIds []string
+	}{
+		{
+			Name:            "Get files with nil option",
+			Page:            1,
+			PerPage:         10,
+			Opt:             nil,
+			ExpectedFileIds: []string{file1_1.Id, file1_2.Id, file1_3.Id, file2_1.Id},
+		},
+		{
+			Name:            "Get files including deleted",
+			Page:            1,
+			PerPage:         10,
+			Opt:             &model.GetFileInfosOptions{IncludeDeleted: true},
+			ExpectedFileIds: []string{file1_1.Id, file1_2.Id, file1_3.Id, file2_1.Id, file2_2.Id},
+		},
+		{
+			Name:    "Get files including deleted filtered by channel",
```

Comment:
Can we also test filtering by user?

Reply:
There was already a test for filtering by user a few lines below. I've added another test to test filtering by user and channel.

Reply:
Whoops! I thought I looked carefully, but perhaps not -- thanks for the bonus test anyway :)